lib/draper/decoratable.rb

```diff
@@ -69,16 +69,16 @@ def decorator_class?
       # `Product` maps to `ProductDecorator`).
       #
       # @return [Class] the inferred decorator class.
-      def decorator_class
+      def decorator_class(called_on = self)
```

Comment:
@pascalbetz Thanks for the contribution! I am not crazy about a public(ish) method taking a parameter that is only relevant to draper's internals. However, I can't think of a better way to accomplish this and it definitely solves a longstanding (and very annoying) problem.

Can you `rebase` with master before I merge? There has been some changes to this method in master as well and want to make sure the build passes. After that, I think we are good to go.

Reply:
@codebycliff I understand.
This was the easiest way to solve the problem... Alternatively we could change from recursion to iteration, this would get rid of the parameter. WDYT?

Reply:
@pascalbetz It looks like however you merged, it brought in `master` changes to this pull request... Can you try again with `rebase`? 

As far as the solution goes, I'm not sure. I'm not super opposed to this solution, however, I am interested in what the iteration solution looks like. Especially now that `decorator_class` returns  `nil` if can't be found. It might make the iteration easier.